src/v/cluster_link/tests/deps.h

```diff
@@ -330,6 +343,75 @@ class fake_topic_metadata_cache
       _topic_cfgs;
 };
 
+class fake_topic_creator : public kafka::data::rpc::topic_creator {
```

Comment:
looks like a straight copy of `kafka/data/rpc/test/deps.h:fake_topic_creator`... why can't we use that?